### PR TITLE
Wide deep recsys

### PR DESCRIPTION
**Pull request by jrzaurin**
1. Added example scripts and notebooks on how to use the library in the context of recommendation systems using [this notebook](https://www.kaggle.com/code/matanivanov/wide-deep-learning-for-recsys-with-pytorch) as example. This is a response to issue #133
2. Use the opportunity to add the movilens 100k dataset to the library, so that now it can be imported from the `datasets` module
3. Added a simple (not pre-trained) transformer model to to the text component
4. Added citation file
5. Fix a bug regarding the padding index not being 1 when using the fastai transforms

### CODE REVIEW

**Comment by jrzaurin** [comment]
Needs to be tested on GPU

**Comment by jrzaurin** [comment]
tested on GPU
